core/src/main/java/io/questdb/griffin/SqlOptimiser.java

```diff
@@ -4668,16 +4668,25 @@ && isHourKeyword(qc.getAst().token) && qc.getAst().paramCount == 1 && qc.getAst(
             }
         }
 
-        // check if translating model is redundant, e.g.
-        // that it neither chooses between tables nor renames columns
-        boolean translationIsRedundant = (useInnerModel || useGroupByModel || useWindowModel) && !forceTranslatingModel;
-        if (translationIsRedundant) {
-            for (int i = 0, n = translatingModel.getBottomUpColumns().size(); i < n; i++) {
-                QueryColumn column = translatingModel.getBottomUpColumns().getQuick(i);
-                if (!column.getAst().token.equals(column.getAlias())) {
-                    translationIsRedundant = false;
-                    break;
-                }
+
+        boolean overrideTranslation = false;
```

Comment:
This variable seems redundant.

Reply:
I mean that on L4673 you can do:
```java
boolean translationIsRedundant = checkIfTranslatingModelIsRedundant(useInnerModel, useGroupByModel, useWindowModel, forceTranslatingModel, false, translatingModel);
```

and L4689 could be:
```java
translationIsRedundant = checkIfTranslatingModelIsRedundant(useInnerModel, useGroupByModel, useWindowModel, forceTranslatingModel, true, translatingModel);
```

This way you don't need to introduce the `overrideTranslation` variable. Am missing something?

Reply:
Yes, we're using IntelliJ IDEA - it has a nice built-in linter and auto-formatter, even in the free CE version.